### PR TITLE
Colinks Make Profiles and other pages public

### PR DESCRIPTION
**Pull request by crabsinger**
# Enable anon users to view profiles and other explore page data

## TODO:
- [x] BuyOrSellCoLinks should have a CTA and some way to login or button to click
- [x] Explore and Explore skills pages should be moved and made public

### CODE REVIEW

**Comment by wingfeatherwave** [comment]
made a couple changes. 
- logo should point to /explore if public 
- log in cta is a button now
 
Screenshot 2024-07-30 at 4 18 44 PM

File: hasura/metadata/databases/default/tables/public_farcaster_accounts.yaml [hunk 1]
```diff
@@ -4,13 +4,7 @@ table:
...
   - name: profile_public
     using:
-      manual_configuration:
-        column_mapping:
-          profile_id: id
-        insertion_order: null
-        remote_table:
-          name: profiles_public
-          schema: public
+      foreign_key_constraint_on: profile_id
```
Comment (CryptoGraffe) [inline hasura/metadata/databases/default/tables/public_farcaster_accounts.yaml:7]: I feel like we just flipped this in the other dir - whats going on here? I might have messed this up

File: src/features/colinks/CoLinksContext.tsx [hunk 2]
```diff
@@ -64,26 +64,28 @@ const CoLinksProvider: React.FC<CoLinksProviderProps> = ({ children }) => {
...
-        } else if (!data.profile.tos_agreed_at) {
-          navigate(coLinksPaths.wizard);
-        } else if (!data.profile.cosoul) {
-          // show the mint button
-          navigate(coLinksPaths.wizard);
-        } else if (!data.profile.links_held) {
-          // redirect to wizard so they can buy their own link
-          // we might already be on the wizard
-          if (location.pathname !== coLinksPaths.wizard) {
+    if (address) {
```
Comment (CryptoGraffe) [inline src/features/colinks/CoLinksContext.tsx:67]: This change is the hairiest to me. I also don't see this working - If I visit /home it takes me to /start

Comment (CryptoGraffe) [reply src/features/colinks/CoLinksContext.tsx:67]: i.e. there is no way to proceed without connecting wallet

File: src/features/colinks/Mutes.tsx [hunk 3]
```diff
@@ -20,12 +20,13 @@ export const Mutes = ({
   targetProfileId: number;
   targetProfileAddress: string;
 }) => {
-  const profileId = useAuthStore(state => state.profileId);
+  const profileId = useProfileId(false);
```
Comment (CryptoGraffe) [inline src/features/colinks/Mutes.tsx:23]: we just shouldn't show this whole component if not logged in

File: src/features/colinks/useDoWithCoLinksContract.ts [hunk 4]
```diff
@@ -38,6 +38,11 @@ export const useDoWithCoLinksContract = () => {
       return;
     }
 
+    if (!authAccount) {
+      showError('Please connect your wallet');
```
Comment (CryptoGraffe) [inline src/features/colinks/useDoWithCoLinksContract.ts:42]: This should probably do something nicer like popup the wallet modal when they interact?

File: src/features/colinks/CoLinksContext.tsx (hunk 2 above)
Comment (CryptoGraffe) [reply src/features/colinks/CoLinksContext.tsx:67]: i figured out i can test with /explore

File: hasura/metadata/databases/default/tables/public_farcaster_accounts.yaml (hunk 1 above)
Comment (crabsinger) [reply hasura/metadata/databases/default/tables/public_farcaster_accounts.yaml:7]: keeping what we have on main.

File: src/features/colinks/CoLinksContext.tsx [hunk 5]
```diff
@@ -99,13 +101,13 @@ const CoLinksProvider: React.FC<CoLinksProviderProps> = ({ children }) => {
...
-  if (data === undefined) {
+  if (isLoading) {
     return <LoadingModal visible={true} />;
   }
 
-  if (!data) {
-    return <Text>Loading...</Text>;
-  }
+  // if (!data) {
+  //   return <Text>Loading...</Text>;
```
Comment (crabsinger) [inline src/features/colinks/CoLinksContext.tsx:109]: i think we can remove this... not sure if theres an error case here we are not handling tho

File: src/features/colinks/Mutes.tsx (hunk 3 above)
Comment (crabsinger) [reply src/features/colinks/Mutes.tsx:23]: hmm, lets discuss.

File: src/features/colinks/useDoWithCoLinksContract.ts (hunk 4 above)
Comment (crabsinger) [reply src/features/colinks/useDoWithCoLinksContract.ts:42]: Yeah, I don't know if we have a nice pattern for this right now.

**Pull request merged**